modules/compiler/vecz/source/analysis/uniform_value_analysis.cpp

```diff
@@ -76,6 +77,32 @@ bool isDivergenceReduction(const Function &F) {
           L.Consume("divergence_"));
 }
 
+bool isTrueUniformInternal(const Value *V, unsigned Depth) {
+  // Arguments and Constants are always uniform
+  if (isa<Argument>(V) || isa<Constant>(V)) {
+    return true;
+  }
+
+  constexpr unsigned DepthLimit = 6;
+
+  if (Depth < DepthLimit) {
+    // For a specific subset of instructions, if all operands are truly
+    // uniform, then the instruction is too.
+    // FIXME: This is pessimistic.
```

Comment:
I wonder why this is pessimistic - could you give an example/explanation here?

Reply:
"conservative" might be a better term. Just because we're not handling things like intrinsics, calls, loads, stores, allocas, vector insert/extract, etc. It's hard to get 100% full coverage here.

Reply:
oh i see it's the specific subset that is pessimistic - maybe say that in the FIXME: This is pessimistic, improve by covering a larger set of instructions?